src/components/search/search.tsx

```diff
@@ -59,10 +127,56 @@ const Search: React.FunctionComponent<SearchProps> = ({
         }
     };
 
+    const onLoadMore = (e: React.MouseEvent<HTMLAnchorElement>) => {
+        e.preventDefault(); // If JS is enabled, do not follow href.
+
+        // If search query and filters are empty, then assume
+        // we are traversing all content with pagination.
+        if (hasEmptyFilterAndQuery()) {
+            const nextPage = currentPage + 1;
+
+            setCurrentPage(nextPage);
+            updatePageTitle(nextPage);
+
+            // Need to pass slug for dynamic pages.
+            // https://github.com/vercel/next.js/discussions/8207
+            const query = pageSlug
+                ? {
+                      slug: pageSlug.slice(1),
+                      page: nextPage,
+                  }
+                : { page: nextPage };
+
+            router.replace(
+                {
+                    pathname: router.pathname,
+                    query: query,
+                },
+                undefined,
+                {
+                    scroll: false,
+                    shallow: true,
+                }
+            );
+            setResultsToShow(currentPage * DEFAULT_PAGE_SIZE + 10);
```

Comment:
After refactoring, this will be replaced with `setSize` from useSWRInfinite. The router handling and page number tracking will still be needed though for SEO.